src/utils/helpers.js

```diff
@@ -1,165 +1,194 @@
-import { ALLOWED_TAGS, TAGS_TO_COMPONENT_MAP, IS_INLINE_COMPONENT } from "./constant"
+import {
+  ALLOWED_TAGS,
+  TAGS_TO_COMPONENT_MAP,
+  IS_INLINE_COMPONENT,
+} from "./constant";
 
 //Vimeo and youtube validator
 export const getVideoUrl = (url) => {
-	let videoid = url.replace(/(https{0,1}:\/\/){0,1}(www\.){0,1}((youtube.com\/{0,1}(watch\?v=){0,1})|(vimeo.com\/{0,1}))/g, "")
-	if (url.includes('vimeo')) {
-		return `//player.vimeo.com/video/${videoid}`
-	}
-	else if (url.includes('youtube')) {
-		return `https://www.youtube.com/embed/${videoid}`
-	}
-	return ''
-}
-
-  // Method to position the cursor at the end of the content
+  let videoid = url.replace(
+    /(https{0,1}:\/\/){0,1}(www\.){0,1}((youtube.com\/{0,1}(watch\?v=){0,1})|(vimeo.com\/{0,1}))/g,
+    ""
+  );
+  if (url.includes("vimeo")) {
+    return `//player.vimeo.com/video/${videoid}`;
+  } else if (url.includes("youtube")) {
+    return `https://www.youtube.com/embed/${videoid}`;
+  }
+  return "";
+};
+
+export const urlify = (text) => {
+  var urlRegex = /(((https?:\/\/)|(www\.))[^\s]+)/g;
+  return text.replace(urlRegex, function (url, b, c) {
+    let url2 = c == "www." ? "http://" + url : url;
+    return '<a href="' + url2 + '" target="_blank">' + url + "</a>";
```

Comment:
@Sathyanarayan09 bro check this function once more know? I have a feeling this can be done differently